pkg/batcher/describeinstances.go

```diff
@@ -61,41 +62,52 @@ func FilterHasher(ctx context.Context, input *ec2.DescribeInstancesInput) uint64
 	return hash
 }
 
-func execDescribeInstancesBatch(ec2api ec2iface.EC2API) BatchExecutor[ec2.DescribeInstancesInput, ec2.DescribeInstancesOutput] {
+func execDescribeInstancesBatch(ec2api sdk.EC2API) BatchExecutor[ec2.DescribeInstancesInput, ec2.DescribeInstancesOutput] {
 	return func(ctx context.Context, inputs []*ec2.DescribeInstancesInput) []Result[ec2.DescribeInstancesOutput] {
 		results := make([]Result[ec2.DescribeInstancesOutput], len(inputs))
 		firstInput := inputs[0]
 		// aggregate instanceIDs into 1 input
 		for _, input := range inputs[1:] {
 			firstInput.InstanceIds = append(firstInput.InstanceIds, input.InstanceIds...)
 		}
-		missingInstanceIDs := sets.NewString(lo.Map(firstInput.InstanceIds, func(i *string, _ int) string { return *i })...)
+		missingInstanceIDs := sets.NewString(lo.Map(firstInput.InstanceIds, func(i string, _ int) string { return i })...)
+		paginator := ec2.NewDescribeInstancesPaginator(ec2api, firstInput)
+
+		for paginator.HasMorePages() {
+			output, err := paginator.NextPage(ctx)
+			// if err != nil {
+			// 	for i := range results {
+			// 		if inputs[i].InstanceIds[0] == *instance.InstanceId {
+			// 			results[i] = Result[ec2.DescribeInstancesOutput]{Err: err}
+			// 		}
+			// 	}
+			// }
+			if err != nil {
+				break
```

Comment:
The Paginator errors when it is out of pages or when describe instances fails. In the event it fails for that page we still need to go through the rest of the pages. When results is returned it acknowledges that there was an error with that specific call. However if we don't break here it continues to line 89 where reservations does not exist because describeinstances failed.